packages/fuels-core/src/lib.rs

```diff
@@ -105,7 +134,7 @@ impl Tokenizable for bool {
     }
 }
 
-impl Tokenizable for String {
+impl Tokenizable for StringToken {
     fn from_token(token: Token) -> Result<Self, Error> {
         match token {
             Token::String(data) => Ok(data),
```

Comment:
Here I think you should explicit that `data` is actually a tuple (iiuc), so it's less confusing.

Reply:
It is a `struct ` but I got your point. We could put `data @ StringToken { .. }` but IMO it is redundant as we are implementing it for `StringToken`. I changed `data` to `string_token` and I think it is clear what is done here.